prow/crier/controller.go

```diff
@@ -199,9 +208,14 @@ func (r *reconciler) reconcile(ctx context.Context, log *logrus.Entry, req recon
 		return requeue, nil
 	}
 
-	log.Info("Reported job(s), now will update pj(s)")
+	log.Infof("Reported job(s), now will update %d pj(s)", len(pjs))
 	for _, pjob := range pjs {
-		if err := r.updateReportStateWithRetries(ctx, pjob, log); err != nil {
+		// Spend up to 10 mins attempting to update report state since the behavior
+		// is poor if we fail. This long of a delay is needed for when the API
+		// server is temporarily unavailable.
+		updateContext, updateCancel := context.WithTimeout(ctx, 10*time.Minute)
+		defer updateCancel()
```

Comment:
Queuing up 10,000 `defer updateCancel()` call to run when we return seems a little excessive. Consider just a single one of these outside the for loop.